test/custom-parser.test.js

```diff
@@ -673,6 +673,113 @@ test('Should get the body as string', t => {
   })
 })
 
+test('Should return defined body with no custom parser defined and content type = \'text/plain\'', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.post('/', (req, reply) => {
+    reply.send(req.body)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    sget({
+      method: 'POST',
+      url: 'http://localhost:' + fastify.server.address().port,
+      body: 'hello world',
+      headers: {
+        'Content-Type': 'text/plain'
+      }
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(body.toString(), 'hello world')
+      fastify.close()
+    })
+  })
+})
+
+test('Should have typeof body string with no custom parser defined, no body defined and content type = \'text/plain\'', t => {
+  t.plan(4)
+  const fastify = Fastify()
+
+  fastify.post('/', (req, reply) => {
+    reply.send(req.body)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    sget({
+      method: 'POST',
+      url: 'http://localhost:' + fastify.server.address().port,
+      headers: {
+        'Content-Type': 'text/plain'
+      }
+    }, (err, response, body) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(typeof (body.toString()), 'string')
```

Comment:
On top off my head, `typeof (body.toString())` does not make sense 🤔  

it should be `typeof body` only probably?

Reply:
Yes, that was a bit of a dead end as the body at that point was always a buffer object.

I'll remove .toString() and test for type object.

Reply:
@lfurzewaddock maybe I am wrong though, lets wait for @mcollina

Reply:
done.